docker/docker_image_src.go

```diff
@@ -236,13 +236,23 @@ func (s *dockerImageSource) ensureManifestIsLoaded(ctx context.Context) error {
 	return nil
 }
 
-func (s *dockerImageSource) getExternalBlob(ctx context.Context, urls []string) (io.ReadCloser, int64, error) {
+func supportedHTTPURLs(urls []string) (supported []string) {
+	for _, s := range urls {
+		if u, err := url.Parse(s); err == nil && (u.Scheme == "http" || u.Scheme == "https") {
+			supported = append(supported, s)
+		}
+	}
+	return
+}
+
+func (s *dockerImageSource) getExternalBlob(ctx context.Context, urls []string) (_ io.ReadCloser, _ int64, fallback bool, _ error) {
+	urls = supportedHTTPURLs(urls)
 	var (
 		resp *http.Response
 		err  error
 	)
 	if len(urls) == 0 {
-		return nil, 0, errors.New("internal error: getExternalBlob called with no URLs")
+		return nil, 0, true, errors.New("internal error: getExternalBlob called with no URLs")
 	}
 	for _, url := range urls {
```

Comment:
I think the url check should be done here.  This way, we iterate over the `urls` slice once but not twice.

Reply:
Thank you for the comment. Fixed.